lib/fastly_nsq/rake_task.rb

```diff
@@ -3,36 +3,71 @@
 
 module MessageQueue
   class RakeTask < Rake::TaskLib
-    attr_accessor :name, :topic, :channel
+    attr_accessor :name, :channel
 
     def initialize(*args, &task_block)
       @name = args.shift || :begin_listening
 
-      desc 'Listen to NSQ on topic using channel' unless ::Rake.application.last_comment
+      unless ::Rake.application.last_comment
+        desc 'Listen to NSQ on topic using channel'
+      end
 
       task(name, *args) do |_, task_args|
         RakeFileUtils.send(:verbose, verbose) do
-          yield(*[self, task_args].slice(0, task_block.arity)) if block_given?
-          @topic   = task_args[:topic] if task_args[:topic]
-          @channel = task_args[:channel] if task_args[:channel]
-          run_task
+          if block_given?
+            yield(*[self, task_args].slice(0, task_block.arity))
+          end
+
+          if task_args[:channel]
+            @channel = task_args[:channel]
+          end
+
+          guard_missing_channel
```

Comment:
I like to guard early for missing required args. Potentially the block above this could also throw an exception because of a bad or missing channel.